### ISSUE TITLE
Current user cannot access cda files

### ISSUE THREAD

**Issue by LuisMaria** [Mgiepz/saiku-adhoc#189]
Hi,
I've installed pentaho 4.8 and the plugin saiku-adhoc-plugin-1.0-GA.

When I try to view a report with an user without Admin role, the report show the datasources, columns, groups and filters defined, but no data.

In the same way, if I try to create a new report with an user without Admin role, I can select a datasource, but when I place one of the fields in the report to act as a column, the error message "ReportGenerator.ERROR_0001 - The CDA Datafactory could not process the query "appears on screen.

In both cases, in the log file (pentaho.log) appears errors about access permissions to generated cda files:

2015-03-30 12:23:08,738 WARN  [pt.webdetails.cda.CdaContentGenerator] Parameter 'httpresponse' not found!
2015-03-30 12:23:08,741 WARN  [pt.webdetails.cpf.repository.pentaho.PentahoLegacySolutionAccess] hasAccess: cda extension not in acl-files.
2015-03-30 12:23:08,741 WARN  [pt.webdetails.cpf.repository.pentaho.PentahoLegacySolutionAccess] hasAccess: Unable to check access control for /system/tmp/C9D83B1C-5F79-DFDD-AAB5-C8BC2B20DA86.cda using default access settings.
2015-03-30 12:23:08,742 ERROR [pt.webdetails.cda.CdaContentGenerator] listParameters: Current user cannot access "/system/tmp/C9D83B1C-5F79-DFDD-AAB5-C8BC2B20DA86.cda"
pt.webdetails.cda.AccessDeniedException: Current user cannot access "/system/tmp/C9D83B1C-5F79-DFDD-AAB5-C8BC2B20DA86.cda"

I have try changing the role who acts how Pentaho Admin (<admin-role> in pentaho.xml) and then, users with the new Admin Role have no probles.

Can help me?

Thanks in advance.

**Comment by johncobley** [comment]
Luis, how did you fix this? I seem to be experiencing the same issue after updating plugins.

**Comment by LuisMaria** [comment]
Hi John, 

May be a bug in cda plugins,  I'm not sure.

I think is related with roles and Spring security permissions.

I haven't fixed it, bat I have found a workaround for our controled environment. We have granted to all users the "cto" role, who have similar permissions than "Admin" role but can't see "tools" in the top menu.
And also whe have manage the permissions over folders and reports with other roles differents of "cto".

Best Regards and Sorry by my English
Luis
